### ISSUE TITLE
Error: Property 'range' of onError parameter is incorrect.

### ISSUE THREAD

**Issue by tomschenkjr** [igorshubovych/markdownlint-cli#95]
A project contains around 30 documents, but a single document throws an error:

```bash
/usr/local/lib/node_modules/markdownlint-cli/node_modules/markdownlint/lib/markdownlint.js:799
      throw error;
      ^

Error: Property 'range' of onError parameter is incorrect.
    at throwError (/usr/local/lib/node_modules/markdownlint-cli/node_modules/markdownlint/lib/markdownlint.js:502:13)
    at onError (/usr/local/lib/node_modules/markdownlint-cli/node_modules/markdownlint/lib/markdownlint.js:531:9)
    at addError (/usr/local/lib/node_modules/markdownlint-cli/node_modules/markdownlint/helpers/helpers.js:433:3)
    at addErrorDetailIf (/usr/local/lib/node_modules/markdownlint-cli/node_modules/markdownlint/helpers/helpers.js:447:5)
    at /usr/local/lib/node_modules/markdownlint-cli/node_modules/markdownlint/lib/md044.js:44:19
    at Array.forEach (<anonymous>)
    at forToken (/usr/local/lib/node_modules/markdownlint-cli/node_modules/markdownlint/lib/md044.js:25:12)
    at forChild (/usr/local/lib/node_modules/markdownlint-cli/node_modules/markdownlint/helpers/helpers.js:316:9)
    at Array.forEach (<anonymous>)
    at forToken (/usr/local/lib/node_modules/markdownlint-cli/node_modules/markdownlint/helpers/helpers.js:314:20)
```

This has been reproduced on markdownlint-cli v0.22.0 and v0.23.0 for both Windows 10 and Docker's [Slim image](https://github.com/docker-slim/docker-slim) (above output from v0.23 on Slim). Each file in the project was tested to isolate the problem to this one particular file.

Unfortunately, this file contains sensitive information, is fairly long (~500 lines), and has a wide mix of Markdown content (tables, img links, lists, etc.). It would be hard to clean-up and share without changing the consequence. However, I wonder if you can help point to a likely cause based on the message (e.g., table, ol, ul, etc.). Unfortunately, there is not a reference to a rule to help pin-down a potential cause.

From there, I can try to narrow down on particular sections, attempt to isolate, and report on what is found to be causing the error through local testing.

**Comment by nschonni** [comment]
Could be that you're running an older Node version like https://github.com/igorshubovych/markdownlint-cli/issues/90

**Comment by tomschenkjr** [comment]
Currently running Nodejs 14.2.0.

The older version of Node.js seems associated with `content.trimEnd` errors (something I encountered with Node.js 8.x with markdownlint-cli 0.22.0).

**Comment by DavidAnson** [comment]
The process of narrowing it down may be tedious. Would it be possible for you to do something like replacing all the letters in the document with “X“ and then sharing it with me? That should completely obscure any confidential data without affecting the scenario.

**Comment by DavidAnson** [comment]
Actually, the stack trace tells us that the problem is with rule MD044: https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md044---proper-names-should-have-the-correct-capitalization

That rule will only suggest fixes or violations based on a set of proper names you’ve provided it in configuration. Therefore, you may be able to search for all instances of those proper names in the file in order to fairly quickly narrow down the issue.

**Comment by tomschenkjr** [comment]
Thanks, this worked well. Found the proper name "Microsoft" in this line:

> Using domain accounts to connect to SQL Server from Databricks results in an authentication error such as `com.microsoft.sqlserver.jdbc.SQLServerException:`

Thanks for the pointer to narrow this down--save a ton of time. 

But have an additional question for MD044. Even when `code_blocks: false`, the code span seems to be included in the evaluation of proper names. Is there a way to exclude code span from MD044?

In the meantime, I've changed the code span to a code block as a fix

**Comment by nschonni** [comment]
@DavidAnson I found a public repo that is hitting the same issue with MD044 https://github.com/MicrosoftDocs/PowerShell-Docs

**Comment by DavidAnson** [comment]
@tomschenkjr Testing from my web browser, I’m not sure that one line is enough to reproduce the crash. It seems fine in the web demo: https://dlaa.me/markdownlint/#%25m%23%20Issue%2095%0A%0AUsing%20domain%20accounts%20to%20connect%20to%20SQL%20Server%20from%20Databricks%20results%20in%20an%20authentication%20error%20such%20as%20%60com.microsoft.sqlserver.jdbc.SQLServerException%3A%60%0A

Can you please check if some other part of the file may be necessary?

@nschonni Thanks! Do you know which file? Did you find it by running locally, or is there a CI run to look at?

**Comment by DavidAnson** [comment]
@tomschenkjr The `code_blocks` parameter should already exclude inline code blocks: https://github.com/DavidAnson/markdownlint/blob/ada0d02a2f0c91bc2133fb975c8de66f5381bbc7/lib/md044.js#L64

This makes me even more curious about what surrounding content may be playing a role in your scenario.

**Comment by tomschenkjr** [comment]
I think I found the contributing issue. The source code was:

>Using domain accounts to connect to SQL Server from Databricks results in an authentication error such as `
> com.microsoft.sqlserver.jdbc.SQLServerException:`

I thought it was just a line wrap, but there is a line break between \` and `com.microsoft.`. That is, this is line 1:

> Using domain accounts to connect to SQL Server from Databricks results in an authentication error such as `

And this is line 2:

> com.microsoft.sqlserver.jdbc.SQLServerException:

After eliminating all other content, was able to isolate the test by running the initial code with the line break which reproduced the error. However, after eliminating the line break, the tests were successfully executed. Thus, it seems the line break immediately after the code span started was causing the error. 

When testing this in the [online editor](https://dlaa.me/markdownlint/#%25m%23%20Issue%2095%0A%0AUsing%20domain%20accounts%20to%20connect%20to%20SQL%20Server%20from%20Databricks%20results%20in%20an%20authentication%20error%20such%20as%20%60com.microsoft.sqlserver.jdbc.SQLServerException%3A%60%0A), I receive the MD038 error and does not replicate the bug that arose in my local instance. 

If there is interest in digging further, we've been doing testing using a Docker container using 
[this Dockerfile file](https://github.com/igorshubovych/markdownlint-cli/files/4623473/Dockerfile.txt) (GitHub doesn't like files without extensions so added .`txt`). Assuming the `markdown.md` with the offending code is placed within `/path/to/local/directory` then the code can be executed with:

```bash
docker run -v /path/to/local/directory:/markdown docker-image ./markdown.md
```

**Comment by nschonni** [comment]
> @nschonni Thanks! Do you know which file? Did you find it by running locally, or is there a CI run to look at?

Just locally, I was looking to see if the repo was clean before proposing the same GitHub Action setup as the other repos

**Comment by DavidAnson** [comment]
This does reproduce in the online editor; we just forgot to configure rule MD044 there. Here is an example that we produce is the exception: https://dlaa.me/markdownlint/#%25m%23%20Issue%2095%0A%0AUsing%20domain%20accounts%20to%20connect%20to%20SQL%20Server%20from%20Databricks%20results%20in%20an%20authentication%20error%20such%20as%20%60%0Acom.microsoft.sqlserver.jdbc.SQLServerException%3A%60%0A%0A%3C!--%20markdownlint-configure-file%20%7B%20%22MD044%22%3A%20%7B%20%22names%22%3A%20%5B%20%22Microsoft%22%20%5D%20%7D%20%7D%20--%3E

**Comment by DavidAnson** [comment]
This is fixed by DavidAnson/markdownlint@87aac66 for the next (patch) release of the library/CLI.

Thanks for the great issue!

**Comment by DavidAnson** [comment]
Fixed in 0.23.1
